ezyfox-server-nio/src/main/java/com/tvd12/ezyfoxserver/nio/socket/EzyNioSecureSocketChannel.java

```diff
@@ -0,0 +1,289 @@
+                        handshakeStatus = result.getHandshakeStatus();
+                    } catch (SSLException e) {
+                        engine.closeOutbound();
+                        handshakeStatus = engine.getHandshakeStatus();
+                        logger.info(
+                            "A problem was encountered while processing the data " +
+                                "that caused the SSLEngine to abort. " +
+                                "Will try to properly close connection...",
+                            e
+                        );
+                        break;
+                    }
+                    switch (result.getStatus()) {
+                        case BUFFER_OVERFLOW:
+                            netBuffer = enlargeBuffer(netBuffer, netBufferSize);
+                            break;
+                        case BUFFER_UNDERFLOW:
+                            throw new SSLException("Buffer underflow occurred after a wrap.");
+                        case CLOSED:
+                            try {
+                                writeOrTimeout(channel, netBuffer, endTime);
+                                peerNetData.clear();
+                            } catch (Exception e) {
+                                logger.info(
+                                    "Failed to send server's close message " +
+                                        "due to socket channel's failure.",
+                                    e
+                                );
+                                handshakeStatus = engine.getHandshakeStatus();
+                            }
+                            break;
+                        default: // OK
+                            writeOrTimeout(channel, netBuffer, endTime);
+                            break;
+                    }
+                    break;
+                default: // NEED_TASK:
+                    Runnable task;
+                    while ((task = engine.getDelegatedTask()) != null) {
+                        task.run();
+                    }
+                    handshakeStatus = engine.getHandshakeStatus();
+                    break;
+            }
+        }
+        handshaked = true;
+    }
+
+    public byte[] read(ByteBuffer buffer) throws Exception {
+        if (netBuffer.position() > 0) {
+            netBuffer.compact();
+        }
+        netBuffer.put(buffer);
+        netBuffer.flip();
+        ByteBuffer tcpAppBuffer = ByteBuffer.allocate(appBufferSize);
+        while (netBuffer.hasRemaining()) {
+            SSLEngineResult result = engine.unwrap(netBuffer, tcpAppBuffer);
+            switch (result.getStatus()) {
+                case BUFFER_OVERFLOW:
+                    netBuffer = enlargeBuffer(netBuffer, appBufferSize);
+                    break;
+                case BUFFER_UNDERFLOW:
+                    tcpAppBuffer = enlargeBufferIfNeed(tcpAppBuffer, netBufferSize);
```

Comment:
```java
public static final [SSLEngineResult.Status](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/javax/net/ssl/SSLEngineResult.Status.html) BUFFER_UNDERFLOW
The SSLEngine was not able to unwrap the incoming data because there were not enough source bytes available to make a complete packet.
Repeat the call once more bytes are available.
```
Should return byte[0]?